providers/keap/write.go

```diff
@@ -0,0 +1,83 @@
+package keap
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) {
+	if err := config.ValidateParams(); err != nil {
+		return nil, err
+	}
+
+	url, err := c.getWriteURL(config.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	var write common.WriteMethod
+
+	if len(config.RecordId) == 0 {
+		if supportedObjectsByCreate[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Post
+		}
+	} else {
+		if supportedObjectsByUpdatePUT[c.Module.ID].Has(config.ObjectName) {
```

Comment:
The update is done either via PUT or PATCH. Since I manually put the object names into the sets there is no object present in both. I will leave a comment about this.